source/module_hamilt_pw/hamilt_pwdft/kernels/cuda/ekinetic_op.cu

```diff
@@ -13,16 +13,31 @@ template <typename FPTYPE>
 __global__ void ekinetic_pw(
     const int npw,
     const int max_npw,
+    const bool is_first_node,
     const FPTYPE tpiba2,
     const FPTYPE* gk2,
     thrust::complex<FPTYPE>* hpsi,
     const thrust::complex<FPTYPE>* psi)
 {
   const int block_idx = blockIdx.x;
   const int thread_idx = threadIdx.x;
-  for (int ii = thread_idx; ii < npw; ii+= blockDim.x) {
-    hpsi[block_idx * max_npw + ii]
-      += gk2[ii] * tpiba2 * psi[block_idx * max_npw + ii];
+  if(is_first_node)
+  {
+      for (int ii = thread_idx; ii < npw; ii += blockDim.x)
+      {
+          hpsi[block_idx * max_npw + ii] = gk2[ii] * tpiba2 * psi[block_idx * max_npw + ii];
+      }
+      for (int ii = npw + thread_idx; ii < max_npw; ii += blockDim.x)
+      {
+          hpsi[block_idx * max_npw + ii] = 0.0;
```

Comment:
it seems that block_idx * max_npw can be calculated only once outside of the loop